### PR TITLE
Add tcpout configuration for Splunk forwarders.

### PR DESCRIPTION
**Pull request by gaylatea**
### What does this PR do?
This PR adds some new configuration we discovered today that allows us to directly interface with the Splunk forwarders.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/observability_pipelines/setup/splunk.md [hunk 1]
```diff
@@ -338,16 +338,32 @@ sources:
     address: 0.0.0.0:8088
     valid_tokens:
         - $${SPLUNK_TOKEN}
+  splunk_tcpout:
+    type: socket
+    mode: tcp
+    address: 0.0.0.0:8089
```
Comment (sghall) [inline content/en/observability_pipelines/setup/splunk.md:344]: Wait. The `socket` source was removed from OPW.

**Review by jszwedko — APPROVED** [review]
I like it. 👍 to keeping both the HEC and tcpout sources.

**Review by sghall — CHANGES_REQUESTED** [review]

File: content/en/observability_pipelines/setup/splunk.md (hunk 1 above)
Comment (silversupreme) [reply content/en/observability_pipelines/setup/splunk.md:344]: Ah, that's right. I have an older image of OPW, it seems, that it hasn't been removed from. We should maybe push back at this now that we have a use case that really requires it.

Comment (leebenson) [reply content/en/observability_pipelines/setup/splunk.md:344]: @silversupreme, I've added you as an editor to the [OPW Security Risks](https://docs.google.com/document/d/1GjAFWv2xlb-yNd58tpkKgKkKzPUw_2ZwkdkpkhFlug4/edit?usp=sharing) doc. 

Could you write up a blurb arguing for `socket`, and then tag @arbll, @barieom and OCR to open for discussion?

**Comment by drichards-87** [comment]
@silversupreme I've added the Work in Progress label to this PR to remove it from our On Call queue. Please go ahead and remove it once the PR is ready to be merged. Thanks!

**Pull request closed without merging**